exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/store/LocalPersistentStore.java

```diff
@@ -1,220 +1,377 @@
+    if (enableArchiving == true ) {
+      this.archivalThreshold = drillConfig.getInt(ExecConstants.PROFILES_STORE_CAPACITY);
+      this.archivalRate = drillConfig.getInt(ExecConstants.PROFILES_STORE_ARCHIVE_RATE);
+      this.pendingArchivalSet = new ProfileSet(archivalRate);
+      this.archivePath = new Path(basePath, ARCHIVE_LOCATION);
+    }
+
+    // Timing
+    this.listAndBuildWatch = Stopwatch.createUnstarted();
+    this.transformWatch = Stopwatch.createUnstarted();
+    this.archiveWatch = Stopwatch.createUnstarted();
+
+    // One time transformer function instantiation
+    this.transformer = new Function<String, Entry<String, V>>() {
+      @Nullable
+      @Override
+      public Entry<String, V> apply(String key) {
+        return new ImmutableEntry<>(key, get(key));
+      }
+    };
+
+    //Base Dir
+    try {
+      if (!mkdirs(basePath)) {
+        version++;
+      }
+      //Creating Archive if required
+      if (enableArchiving) {
+        try {
+          if (!fs.exists(archivePath)) {
+            mkdirs(archivePath);
+          }
+        } catch (IOException e) {
+          logger.warn("Disabling profile archiving due to failure in creating profile archive {} : {}", archivePath, e);
+          this.enableArchiving = false;
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Failure setting pstore configuration path.");
+    }
+  }
+
+  protected Path getBasePath() {
+    return basePath;
+  }
+
+  @Override
+  public PersistentStoreMode getMode() {
+    return PersistentStoreMode.PERSISTENT;
+  }
+
+  private boolean mkdirs(Path path) throws IOException {
+    return fs.mkdirs(path);
+  }
+
+  public static Path getLogDir() {
+    String drillLogDir = System.getenv("DRILL_LOG_DIR");
+    if (drillLogDir == null) {
+      drillLogDir = System.getProperty("drill.log.dir");
+    }
+    if (drillLogDir == null) {
+      drillLogDir = "/var/log/drill";
+    }
+    return new Path(new File(drillLogDir).getAbsoluteFile().toURI());
+  }
+
+  public static DrillFileSystem getFileSystem(DrillConfig config, Path root) throws IOException {
+    Path blobRoot = root == null ? getLogDir() : root;
+    Configuration fsConf = new Configuration();
+    if (blobRoot.toUri().getScheme() != null) {
+      fsConf.set(FileSystem.FS_DEFAULT_NAME_KEY, blobRoot.toUri().toString());
+    }
+
+
+    DrillFileSystem fs = new DrillFileSystem(fsConf);
+    fs.mkdirs(blobRoot);
+    return fs;
+  }
+
+  @Override
+  public Iterator<Map.Entry<String, V>> getRange(int skip, int take) {
+    //Marking currently seen modification time
+    long currBasePathModified = 0L;
+    try {
+      currBasePathModified = fs.getFileStatus(basePath).getModificationTime();
+    } catch (IOException e) {
+      logger.error("Failed to get FileStatus for {}", basePath, e);
+      throw new RuntimeException(e);
+    }
+
+    //No need to acquire lock since incoming requests are synchronized

```

Comment:
Could you please point out to the code which proves this statement?

Reply:
The method services HTTP requests, which is reflected in the tests (and graphs above). Hence, the need for a lock seemed unnecessary.